### ISSUE TITLE
Support for generic ephemeral volumes

### ISSUE THREAD

**Issue by PabloPie** [prometheus-operator/prometheus-operator#4315]
**What is missing?**

Support for [generic ephemeral volumes](https://kubernetes.io/docs/concepts/storage/ephemeral-volumes/#generic-ephemeral-volumes). Currently the storageSpec for Prometheus only supports emptyDir and the usual pvc.

**Why do we need it?**

Generic ephemeral volumes are close to emptyDir in that they are ephemeral but they have the advantage of provisioning dedicated volumes if the csi driver supports it, among other things.

**Environment**

* Prometheus Operator version:

    `0.50.0`

**Anything else we need to know?**:

I also tried creating an additional volume for the Prometheus object with VolumeSource ephemeral but I got hit with a validation error:  'unknown field "ephemeral" in com.coreos.monitoring.v1.Prometheus.spec.volumes'.  Even if this worked, there would be no simple way to override the default storage.tsdb.path I presume.

**Comment by simonpasquier** [comment]
>  Even if this worked, there would be no simple way to override the default storage.tsdb.path I presume.

Right. It may be possible once/if we support strategic merge patch for the pod's spec as we do for containers though.

**Comment by PabloPie** [comment]
Would you be interested in adding support for this through the StorageSpec in the Prometheus CRD? I can work on a PR to add
[EphemeralVolumeSource](https://pkg.go.dev/k8s.io/api@v0.22.0/core/v1#EphemeralVolumeSource) as an option.

It would just require an addition similar to https://github.com/prometheus-operator/prometheus-operator/blob/master/pkg/alertmanager/statefulset.go#L125 right?
